presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestRoles.java

```diff
@@ -544,6 +555,79 @@ public void testShowRoles()
+        QueryAssert.assertThat(() -> onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).
+                failsWithMessage(format("Access Denied: Cannot select from table %s", TABLE_NAME));
+    }
+
+    @Test(groups = {HIVE_CONNECTOR, ROLES, AUTHORIZATION, PROFILE_SPECIFIC_TESTS})
+    public void testGrantSelectRevokeAll()
+            throws Exception
+    {
+        onPresto().executeQuery("CREATE ROLE role1");
+        onPresto().executeQuery("GRANT role1 TO USER alice");
+        onPresto().executeQuery(format("CREATE TABLE %s (foo BIGINT)", TABLE_NAME));
+        onPresto().executeQuery(format("GRANT SELECT ON %s TO ROLE role1", TABLE_NAME));
+        QueryAssert.assertThat(onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).hasNoRows();
+        onPresto().executeQuery(format("REVOKE ALL PRIVILEGES ON %s FROM ROLE role1", TABLE_NAME));
+        QueryAssert.assertThat(() -> onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).
+                failsWithMessage(format("Access Denied: Cannot select from table %s", TABLE_NAME));
+    }
+
+    @Test(groups = {HIVE_CONNECTOR, ROLES, AUTHORIZATION, PROFILE_SPECIFIC_TESTS})
+    public void testGrantToRoleRevokeFromUser()
+            throws Exception
+    {
+        onPresto().executeQuery("CREATE ROLE role1");
+        onPresto().executeQuery("GRANT role1 TO USER alice");
+        onPresto().executeQuery(format("CREATE TABLE %s (foo BIGINT)", TABLE_NAME));
+        QueryAssert.assertThat(() -> onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).
+                failsWithMessage(format("Access Denied: Cannot select from table %s", TABLE_NAME));
+        onPresto().executeQuery(format("GRANT SELECT ON %s TO ROLE role1", TABLE_NAME));
+        QueryAssert.assertThat(onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).hasNoRows();
+        onPresto().executeQuery(format("REVOKE SELECT ON %s FROM USER alice", TABLE_NAME));
+        QueryAssert.assertThat(onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).hasNoRows();
+    }
+
+    @Test(groups = {HIVE_CONNECTOR, ROLES, AUTHORIZATION, PROFILE_SPECIFIC_TESTS})
+    public void testGrantToUserRevokeFromRole()
+            throws Exception
+    {
+        onPresto().executeQuery("CREATE ROLE role1");
+        onPresto().executeQuery("GRANT role1 TO USER alice");
+        onPresto().executeQuery(format("CREATE TABLE %s (foo BIGINT)", TABLE_NAME));
+        QueryAssert.assertThat(() -> onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).
+                failsWithMessage(format("Access Denied: Cannot select from table %s", TABLE_NAME));
+        onPresto().executeQuery(format("GRANT SELECT ON %s TO USER alice", TABLE_NAME));
+        QueryAssert.assertThat(onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).hasNoRows();
+        onPresto().executeQuery(format("REVOKE SELECT ON %s FROM ROLE role1", TABLE_NAME));
```

Comment:
Did you intend to grant SELECT to role1 first? Otherwise, this `REVOKE`statement won't do anything, since role1 doesn't have SELECT privilege in the first place.